Automattic/pocket-casts-android
# Update horologist version (and libraries that matches this version) (targeting `main`)

## Description
This PR is a follow-up to https://github.com/Automattic/pocket-casts-android/pull/1634. Instead of targeting the `release/7.54` branch, this PR targets `main` because we were able to extract just the changes we needed for the release and add only those changes to the release branch in https://github.com/Automattic/pocket-casts-android/pull/1636.

In addition, this PR uses horologist version `0.4.17` instead of `0.5.14` because the `0.4.X` versions of horologist use stable androidx APIs whereas `0.5.x` versions use alpha androidx APIs.

This PR also reverts the changes from https://github.com/Automattic/pocket-casts-android/pull/1636 because those changes should not be needed anymore.

> [!WARNING]
> Reverting https://github.com/Automattic/pocket-casts-android/pull/1636 also removed [this change](https://github.com/Automattic/pocket-casts-android/pull/1636#issuecomment-1862192925) you added because you saw the "Contact Support" chip slightly cut off. I didn't observe that issue on this branch so I did not re-add that change, but if you still see the chip being cut off (or you think it's just prudent to add in that extra padding), feel free to add that change back in.

## Testing Instructions

Smoke test the phone, automotive, and especially the watch app for any regressions since this is updating quite a few dependencies.

Also redo the testing steps for https://github.com/Automattic/pocket-casts-android/pull/1636 to make sure that horologist is properly keeping the chips fully visible.

## Checklist
 
#### I have tested any UI changes...
